controllers/managedocs_controller.go

```diff
@@ -255,3 +344,138 @@ func (r *ManagedOCSReconciler) updateComponents() v1.ComponentStatusMap {
 
 	return components
 }
+
+func (r *ManagedOCSReconciler) checkUninstallCondition() bool {
+	cmNamespaceName := types.NamespacedName{
+		Name:      r.DeleteConfigMapName,
+		Namespace: r.namespace,
+	}
+
+	configmap := &corev1.ConfigMap{}
+	err := r.Get(r.ctx, cmNamespaceName, configmap)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false
+		}
+		r.Log.Error(err, "Unable to get configMap")
+		return false
+	}
+	labels := configmap.Labels
+	_, ok := labels[r.DeleteConfigMapLabelKey]
+	return ok
+}
+
+func (r *ManagedOCSReconciler) removeOlmComponents() error {
+
+	r.Log.Info("Deleting subscription")
+	subscription := &operators.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.AddonSubscriptionName,
+			Namespace: r.namespace,
+		},
+	}
+	if err := r.deleteResource(r.Client, subscription, r.AddonSubscriptionName); err != nil {
+		return err
+	}
+	r.Log.Info("subscription removed successfully")
+
+	r.Log.Info("Deleting CSV")
+	csvList := &operators.ClusterServiceVersionList{}
+	err := r.Client.List(r.ctx, csvList)
+	if err != nil {
+		r.Log.Error(err, "Unable to list CSVs")
+		return err
+	}
+	for _, csv := range csvList.Items {
+		if strings.HasPrefix(csv.Name, deployerCsvPrefix) {
+			deployerCsvName := csv.Name
+			csv := &operators.ClusterServiceVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      deployerCsvName,
+					Namespace: r.namespace,
+				},
+			}
+			if err := r.deleteResource(r.Client, csv, deployerCsvName); err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
This loop will not short circuit even after finding the correct CSV. I will suggest separating the finding of the CSV and the action we need to take on that CSV (deleting it)

```suggestion
        var csv operators.ClusterServiceVersion = nil
	for _, candidate := range csvList.Items {
		if strings.HasPrefix(csv.Name, deployerCsvPrefix) {
			csv = candidate
			break
		}
	}
	if err := r.deleteResource(r.Client, csv); err != nil {
		return err
	}
```